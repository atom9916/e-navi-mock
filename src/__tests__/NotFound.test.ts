/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/vue'
import NotFoundVue from '@/views/NotFound.vue'

describe('NotFoundビューのテスト', () => {
  test('エラー画面が正常に表示される', () => {
    render(NotFoundVue)

    const heading = screen.getByRole('heading')
    const text = screen.getByText('アクセスしたページは存在しません')

    expect(heading).toHaveTextContent('Not Found')
    expect(text).toBeInTheDocument()
  })

  test('スナップショットテスト', () => {
    const { container } = render(NotFoundVue)
    expect(container).toMatchSnapshot()
  })
})
