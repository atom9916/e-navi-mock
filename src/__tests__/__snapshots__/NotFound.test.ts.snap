// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundコンポーネント スナップショットテスト 1`] = `
<div>
  
  <h2>
    Not Found
  </h2>
  <p>
    アクセスしたページは存在しません
  </p>
  <div>
    <button>
      ホーム
    </button>
  </div>
  
</div>
`;

exports[`NotFoundビューのテスト スナップショットテスト 1`] = `
<div>
  
  <h2>
    Not Found
  </h2>
  <p>
    アクセスしたページは存在しません
  </p>
  <div>
    <button>
      ホーム
    </button>
  </div>
  
</div>
`;
